### PR TITLE
revamping UI #73

### PR DESCRIPTION
**Pull request by pavanposani**
Update UI to reflect the new changes as per https://github.com/VIDA-NYU/reproserver/issues/73

### CODE REVIEW

**Comment by remram44** [comment]
I deployed it to https://staging.server.reprozip.org/

I think there is a little confusion about what the steps are, the `/web` pages are not replay, they are the web capture, whereas the `/reproduce` pages are not record.

File: reproserver/web/templates/base.html [hunk 1]
```diff
@@ -13,21 +13,22 @@
...
+          <a class="navbar-brand" href="/">ReproServer</a>
             <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbarContent" aria-controls="navbarContent" aria-expanded="false" aria-label="Toggle navigation">
             <span class="navbar-toggler-icon"></span>
           </button>
           <div class="collapse navbar-collapse" id="navbarContent">
-            <ul class="navbar-nav mb-2 mb-md-0 flex-md-grow-1">
-              <li class="nav-item"><a class="nav-link{% if current_nav == 'reproduce' %} active{% endif %}"{% if current_nav == 'reproduce' %} aria-current="true"{% endif %} href="{{ reverse_url('index') }}">Reproduce</a></li>
-              <li class="nav-item"><a class="nav-link{% if current_nav == 'webcapture' %} active{% endif %}"{% if current_nav == 'webcapture' %} aria-current="true"{% endif %} href="{{ reverse_url('webcapture_index') }}">Web capture</a></li>
-              <li class="nav-item ms-md-auto"><a class="nav-link{% if current_nav == 'about' %} active{% endif %}"{% if current_nav == 'about' %} aria-current="page"{% endif %} href="{{ reverse_url('about') }}">About</a></li>
```
Comment (remram44) [inline reproserver/web/templates/base.html:29]: I don't think it is a good idea to remove "reproduce" and "web capture" from the navbar. Having an "upload" option is also misleading because both flows start with an upload form.

File: reproserver/web/templates/index.html [hunk 2]
```diff
@@ -1,51 +1,117 @@
...
-    <label for="rpz_file" class="form-label">Upload a <code>.rpz</code> file</label>
-    <input type="file" class="form-control" id="rpz_file" name="rpz_file" placeholder="experiment.rpz" accept=".rpz">
+<div class="container-fluid shadow p-5 mb-4 bg-light rounded-3">
+  <h1> Share reproducible work </h1>
+  <p>Reprozip allows you to automatically pack your web application along with necessary libraries, configuration
+    files and environment variables</p>
+  <p>Anyone can re run the packed web application on a new machine without tracking and installing the dependencies
+    manually</p>
+  <p>This application is still in progess, Currently It enables you reproduce Reprozip packages from web browser,
+    without having to install reprounzip on your computer</p>
```
Comment (remram44) [inline reproserver/web/templates/index.html]: I don't think the current text needed to be replaced

File: reproserver/web/templates/base.html [hunk 3]
```diff
@@ -13,21 +13,22 @@
   </head>
   <body>
 
-    <div class="container" id="content" role="main">
+    <div class="container-fluid" id="content" role="main">
```
Comment (remram44) [inline reproserver/web/templates/base.html:18]: This forces most of the navigation way off in the corner

File: reproserver/web/templates/index.html [hunk 4]
```diff
@@ -1,51 +1,117 @@
...
+      <div class="col-md-1 text-center mt-5">
+        <svg xmlns="http://www.w3.org/2000/svg" width="64" height="64" fill="currentColor" class="bi bi-arrow-right"
+          viewBox="0 0 16 16">
+          <path fill-rule="evenodd"
+            d="M1 8a.5.5 0 0 1 .5-.5h11.793l-3.147-3.146a.5.5 0 0 1 .708-.708l4 4a.5.5 0 0 1 0 .708l-4 4a.5.5 0 0 1-.708-.708L13.293 8.5H1.5A.5.5 0 0 1 1 8z">
+          </path>
+        </svg>
+      </div>
+      <div class="col">
+        <div class="circle-icon rounded-circle" style="height: 10em; width:10em; opacity: 75%;">
```
Comment (remram44) [inline reproserver/web/templates/index.html]: I think this is slightly off, the text is not centered vertically in the circles for me.

**Review by remram44 — CHANGES_REQUESTED** [review]

File: reproserver/web/templates/index.html (hunk 4 above)
Comment (pavanposani) [reply reproserver/web/templates/index.html]: yes, will fix it in the next commit

File: reproserver/web/templates/base.html (hunk 3 above)
Comment (pavanposani) [reply reproserver/web/templates/base.html:18]: I didn't see any issue in chrome/safari with different viewport options. Could you please elaborate ?

File: reproserver/web/templates/index.html (hunk 2 above)
Comment (pavanposani) [reply reproserver/web/templates/index.html]: https://github.com/VIDA-NYU/reproserver/issues/73  - this mentions the new home page, Please correct me if I'm wrong

Comment (remram44) [reply reproserver/web/templates/index.html]: The problem is that web capture is a work in progress but the rest works. Either way I think we were changing to have the web capture option after the upload, so I guess this will be reworked.

**Comment by remram44** [comment]
The file upload progress doesn't seem to work, I get an error page "405: Method Not Allowed"

File: reproserver/web/__init__.py [hunk 5]
```diff
@@ -51,6 +51,7 @@ def make_app(debug=False, xsrf_cookies=True, proxy=None):
                     name='webcapture_upload_wacz'),
             URLSpec('/web/([^/]+)/download', webcapture.Download,
                     name='webcapture_download'),
+            URLSpec('/record', views.Record, name='record'),
```
Comment (remram44) [inline reproserver/web/__init__.py]: Is there any way to get to this page?

File: reproserver/web/webcapture.py [hunk 6]
```diff
@@ -75,6 +75,10 @@ async def post(self):
 class Dashboard(BaseHandler):
     @PROM_REQUESTS.sync('webcapture_dashboard')
     def get(self, upload_short_id):
+        self.post(upload_short_id=upload_short_id)
+
+    @PROM_REQUESTS.sync('webcapture_dashboard')
+    def post(self, upload_short_id):
```
Comment (remram44) [inline reproserver/web/webcapture.py]: Why is this needed?

File: reproserver/web/templates/setup.html [hunk 7]
```diff
@@ -8,6 +8,7 @@
   <ol class="breadcrumb">
     <li class="breadcrumb-item"><a href="{{ reverse_url('index') }}">Home</a></li>
     <li class="breadcrumb-item active" aria-current="page">Package {{ filename | truncate(60) }}</li>
+    <li class="ms-auto"><a href="{{ reverse_url('webcapture_dashboard', upload_short_id) }}">Web Capture</a></li>
```
Comment (remram44) [inline reproserver/web/templates/setup.html]: I think this shouldn't be in the `breadcrumb` div

File: reproserver/web/templates/record.html [hunk 8]
```diff
@@ -0,0 +1,44 @@
+
+{% set current_nav = 'reproduce' %}
+
+{% extends "base.html" %}
+
+{% block content %}
```
Comment (remram44) [inline reproserver/web/templates/record.html]: Is there any way to get to this page?

File: reproserver/web/templates/setup.html [hunk 9]
```diff
@@ -19,41 +20,43 @@ <h1>Package {{ filename | truncate(60) }}</h1>
 
 <h2>Run the experiment</h2>
 
-<form method="POST" action="{{ reverse_url('start_run', upload_short_id) }}" enctype="multipart/form-data">
+<form method="POST" enctype="multipart/form-data">
```
Comment (remram44) [inline reproserver/web/templates/setup.html]: `action` can probably stay here

File: reproserver/web/__init__.py (hunk 5 above)
Comment (pavanposani) [reply reproserver/web/__init__.py]: will cleanup, not required

File: reproserver/web/templates/record.html (hunk 8 above)
Comment (pavanposani) [reply reproserver/web/templates/record.html]: No, will clean it up

File: reproserver/web/templates/setup.html (hunk 9 above)
Comment (pavanposani) [reply reproserver/web/templates/setup.html]: wouldn't it submit the form twice, because I'm sending an XHR on button click

File: reproserver/web/webcapture.py (hunk 6 above)
Comment (pavanposani) [reply reproserver/web/webcapture.py]: to handle redirects from other pages

Comment (remram44) [reply reproserver/web/webcapture.py]: What do you mean? And wouldn't a redirection use GET?

File: reproserver/web/templates/setup.html (hunk 9 above)
Comment (remram44) [reply reproserver/web/templates/setup.html]: It is still set, you moved it to `formaction` on the button. I don't think that is necessary?

File: reproserver/web/webcapture.py (hunk 6 above)
Comment (pavanposani) [reply reproserver/web/webcapture.py]: If you take the breadcrumbs with `<a href="/web/">` , we have to change them to post with a form

Comment (remram44) [reply reproserver/web/webcapture.py]: You didn't add GET, you added POST. What posts to that endpoint?

File: reproserver/web/templates/base.html [hunk 10]
```diff
@@ -1,66 +1,67 @@
...
+                <li><a class="dropdown-item" href="https://www.reprozip.org/">ReproZip Website</a></li>
+              </ul>
+            </li>
+            <!--end dropdown-->
+          </ul>
+        </div><!-- /.navbar-collapse -->
+      </div><!-- /.container -->
+    </nav><!-- /.navbar -->
 
-      <div class="container mt-3"> <!--Body content-->
```
Comment (remram44) [inline reproserver/web/templates/base.html:45]: This removes the margin between the top navbar and the breadcrumbs

screenshot

Comment (pavanposani) [reply reproserver/web/templates/base.html:45]: fixed in latest commit - 9eccd033e13934b57d254afdce74106739d07f1e

File: reproserver/web/webcapture.py (hunk 6 above)
Comment (pavanposani) [reply reproserver/web/webcapture.py]: this was due to previous design where we have a single form with two buttons and doing submit action, I think we don't need it anymore. removed this change in the latest patchset

File: reproserver/web/templates/setup.html (hunk 9 above)
Comment (pavanposani) [reply reproserver/web/templates/setup.html]: don't need this anymore, this change happened due to previous design, where we have two buttons reproduce/record in the same form. will submit changes in the next commit
